src/client/components/CollaboratorAccessModal/CollaboratorAccessModal.tsx

```diff
@@ -0,0 +1,269 @@
+    options.forEach((option) => {
+      newSelectedSections[option.value] = { ...newSelectedSections[option.value], [permission]: enabled }
+    })
+    setSelectedSections(newSelectedSections)
+  }
+
+  const removeOption = (section: string, permission: string): void => {
+    if (section === 'all') setSelectedSections('none')
+    else if (section === 'none') setSelectedSections('all')
+    else
+      setSelectedSections(
+        typeof selectedSections !== 'string'
+          ? { ...selectedSections, [section]: { ...selectedSections[section], [permission]: false } }
+          : { [section]: { [permission]: false } }
+      )
+  }
+
+  const addOption = (section: string, permission: CollaboratorEditPropertyType): void => {
+    if (section === 'all') setSelectedSections('all')
+    else if (section === 'none') setSelectedSections('none')
+    else
+      setSelectedSections(
+        typeof selectedSections !== 'string'
+          ? { ...selectedSections, [section]: { ...selectedSections[section], [permission]: true } }
+          : { [section]: { [permission]: true } }
+      )
+  }
+
+  const toggleOption = (section: string, permission: CollaboratorEditPropertyType): void => {
+    if (typeof selectedSections === 'string' || !selectedSections[section]?.[permission]) addOption(section, permission)
+    else if (selectedSections[section]?.[permission]) removeOption(section, permission)
+  }
+
+  useEffect(() => {
+    if (open) document.body.classList.add('no-scroll')
+    else document.body.classList.remove('no-scroll')
+  }, [open])
+
+  return (
+    <Modal className="modal-collaborator-access" isOpen={open}>
+      <ModalHeader>
+        {headerLabel}
+        <ModalClose onClose={onClose} />
+      </ModalHeader>
+
+      <ModalBody>
+        <>
+          <div className="form-container inline">
+            <div
+              className="form-field-selector"
+              onClick={() => setSelectedSections('all')}
+              onMouseDown={(e) => e.stopPropagation()}
+              aria-hidden="true"
+            >
+              <div
+                className={classNames('fra-checkbox', {
+                  checked: selectedSections === 'all',
+                })}
+              />
+              <div className="form-field-container-label">{i18n.t(`contactPersons.all`)}</div>
+            </div>
+            <div
+              className="form-field-selector"
+              onClick={() => setSelectedSections('none')}
+              onMouseDown={(e) => e.stopPropagation()}
+              aria-hidden="true"
+            >
+              <div
+                className={classNames('fra-checkbox', {
+                  checked: selectedSections === 'none',
+                })}
+              />
+              <div className="form-field-container-label">{i18n.t(`contactPersons.none`)}</div>
+            </div>
+          </div>
+          <div className="form-container">
+            <MediaQuery maxWidth={Breakpoints.laptop - 1}>
```

Comment:
why two different components? is that really necessary ?

Reply:
I don't know, it's the same on CountrySelectModal. this is how it appears on narrow screens. probably they are unnecessary. we can find another solution for both the modals on a different pr.

<img width="796" alt="Schermata 2022-09-30 alle 09 54 15" src="https://user-images.githubusercontent.com/10047945/193222996-0cff411a-d072-48ef-b095-f7bd5dfe37f0.png">

Reply:
ok